### PR TITLE
load_converter_project

### PR DESCRIPTION
**Pull request by benninoda**

### CODE REVIEW

File: Converter/app/git.gradle [hunk 1]
```diff
@@ -0,0 +1,41 @@
+def getPatch() {
+    def diffOutput = new ByteArrayOutputStream()
+    exec {
+        commandLine 'cmd', 'git', 'describe'
+        standardOutput = diffOutput
+    }
+    def res = diffOutput.toString()
+    if (res.contains('-')) {
```
Comment (gmltA) [inline Converter/app/git.gradle:8]: In which case we will see this condition return `true`?

File: Converter/app/src/main/java/com/javalabs/converter/DataFragment.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        Toast.makeText(getActivity(), R.string.toast_copy, Toast.LENGTH_SHORT).show();
+    }
+
+    public void copyOutput() {
+        ClipboardManager clipboard = (ClipboardManager) getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
+        ClipData clip = ClipData.newPlainText(getString(R.string.output_value), mViewModel.getSelectedDataTo().getValue());
+        clipboard.setPrimaryClip(clip);
+        Toast.makeText(getActivity(), R.string.toast_copy, Toast.LENGTH_SHORT).show();
+    }
+}
```
Comment (gmltA) [inline Converter/app/src/main/java/com/javalabs/converter/DataFragment.java]: According to the requirements, these functions should be present only in one product flavour, so it is better to remove them from the `main` source set (which is used in all flavours).

Then you can create another fragment (just for premium flavour) which will extend this one and enrich it with additional functionality.

Can you do this?)

Comment (benninoda) [reply Converter/app/src/main/java/com/javalabs/converter/DataFragment.java]: I divided DataFragment into different flavors.  I hope this is suitable for your request)

File: Converter/app/git.gradle (hunk 1 above)
Comment (benninoda) [reply Converter/app/git.gradle:8]: I made changes to git. gradle. It turned out that the 'cmd' command was unnecessary and returned the wrong thing. Now it returns true if tag exists

File: Converter/app/src/main/java/com/javalabs/converter/DataFragment.java (hunk 2 above)
Comment (gmltA) [reply Converter/app/src/main/java/com/javalabs/converter/DataFragment.java]: Almost. Now you have a lot of identical code in both fragments.

> Then you can create another fragment (just for premium flavour) which will extend this one and enrich it with additional functionality.

Is it clear what I mean here?

Comment (benninoda) [reply Converter/app/src/main/java/com/javalabs/converter/DataFragment.java]: I made a new fragment, please see if it is correct

**Review by gmltA — APPROVED** [review]
That's better.

**Pull request merged**